pkgs/development/tools/vulkan-validation-layers/default.nix

```diff
@@ -18,6 +19,17 @@ stdenv.mkDerivation rec {
 
   cmakeFlags = [ "-DGLSLANG_INSTALL_DIR=${glslang}" ];
 
+  # Help vulkan-loader find the validation layers
+  setupHook = writeText "setup-hook" ''
```

Comment:
Should this really be a `setupHook`? Layers are runtime dependencies, not build-time ones. There's also a question of host vs. target compatibility.

Reply:
The `setupHook` affects the environment in a `nix-shell`, so if I am building and running a program under development in a `nix-shell`, this lines things up to work.

Reply:
That's what `shellHook` is for. Making changes only intended for shells in the setup hook might have unintended side effects.

Reply:
I found that not to work as I want. Specifically, if you add `vulkan-validation-layers` to your `buildInputs`, then enter a `nix-shell` to work on your program, the environment variable is not updated by `shellHook`. If you want to work on `vulkan-validation-layers` itself, by some mechanism like `nix-shell '<nixpkgs>' -A vulkan-validation-layers`, then, yes, it does get set (though then the use of `@out@` is no longer correct).

As a user, I certainly want the layers found when I take `vulkan-validation-layers` as a dependency, but perhaps I am thinking too narrowly and there is some setup where you only want the layers when working directly on the `vulkan-validation-layers` derivation.

Reply:
Ah, right. Well, I remain concern about how idiomatic this is, but I concede that your approach is eminently useful. Might as well leave it in until someone complains; I suspect validation layers in build environments other than shell environments are a rarity.